community/values/src/main/java/org/neo4j/values/storable/UTF8StringValue.java

```diff
@@ -180,46 +180,60 @@ public int computeHash()
     @Override
     public long updateHash( HashFunction hashFunction, long hash )
     {
-
-        byte[] values = bytes;
-
-        int codePointCount = 0;
-
-        int i = offset;
-        int len = offset + byteLength;
-        while ( i < len )
+        class CodePointCursor
```

Comment:
The code point loop that now exists in this `CodePointCursor` is sort of duplicated elsewhere too. Would it be a good idea to port those places over to using it too? Also such objects will probably be optimized away since they don't escape this method, right?